app/controllers/propertyLinking/ClaimPropertyOccupancyController.scala

```diff
@@ -0,0 +1,149 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.propertyLinking
+
+import actions.AuthenticatedAction
+import actions.propertylinking.WithLinkingSession
+import com.google.inject.Singleton
+import config.ApplicationConfig
+import connectors.SubmissionIdConnector
+import connectors.propertyLinking.PropertyLinkConnector
+import controllers._
+import form.Mappings._
+import models._
+import play.api.Configuration
+import play.api.data.Form
+import play.api.data.Forms._
+import play.api.i18n.{Messages, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepo
+import services.BusinessRatesAttachmentsService
+import services.propertylinking.PropertyLinkingService
+import uk.gov.hmrc.http.UpstreamErrorResponse
+import uk.gov.hmrc.propertylinking.errorhandler.CustomErrorHandler
+import uk.gov.voa.play.form.ConditionalMappings._
+import views.helpers.Errors
+import java.time.LocalDate
+
+import javax.inject.{Inject, Named}
+import org.checkerframework.checker.compilermsgs.qual.CompilerMessageKey
+import utils.Formatters
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ClaimPropertyOccupancyController @Inject()(
+      val errorHandler: CustomErrorHandler,
+      val submissionIdConnector: SubmissionIdConnector,
+      @Named("propertyLinkingSession") val sessionRepository: SessionRepo,
+      authenticatedAction: AuthenticatedAction,
+      withLinkingSession: WithLinkingSession,
+      val propertyLinksConnector: PropertyLinkConnector,
+      businessRatesAttachmentService: BusinessRatesAttachmentsService,
+      val runModeConfiguration: Configuration,
+      occupancyOfPropertyView: views.html.propertyLinking.occupancyOfProperty,
+      serviceUnavailableView: views.html.errors.serviceUnavailable,
+      propertyLinkingService: PropertyLinkingService)(
+      implicit executionContext: ExecutionContext,
+      override val messagesApi: MessagesApi,
+      override val controllerComponents: MessagesControllerComponents,
+      val config: ApplicationConfig
+) extends PropertyLinkingController {
+
+  import ClaimPropertyOccupancy._
+
+  def showOccupancy(): Action[AnyContent] = authenticatedAction.andThen(withLinkingSession).async { implicit request =>
+    {
+
+      propertyLinkingService.findEarliestStartDate(request.ses.uarn).flatMap {
+        case Some(date) if date.isAfter(LocalDate.now()) =>
+          businessRatesAttachmentService
+            .persistSessionData(request.ses.copy(
+              propertyOccupancy = Some(PropertyOccupancy(stillOccupied = true, lastOccupiedDate = None))))
+            .map { _ =>
+              Redirect(routes.ChooseEvidenceController.show())
+            }
+            .recover {
```

Comment:
do we need to recover with a service unavailable view? I think if there's a upstream 5xx response, can't we let that failed future be handled by global error handler? it just displays service unavailable anyways